packages/cli/src/databases/utils/transformers.ts

```diff
@@ -3,8 +3,9 @@ import { ValueTransformer } from 'typeorm';
 import config from '@/config';
 
 export const idStringifier = {
-	from: (value: number): string | number => (typeof value === 'number' ? value.toString() : value),
-	to: (value: string): number | string => (typeof value === 'string' ? Number(value) : value),
+	from: (value: number): string => value?.toString(),
+	to: (value: string | unknown): number | unknown =>
+		typeof value === 'string' ? Number(value) : value,
```

Comment:
Should we try/catch if `Number()` fails?

Reply:
Anything unionized to `unknown` resolves to `unknown`, so this cannot be the type? Also here anything we pass that is not a string will directly end up as-is in the DB?

I'd validate that anything we pass in as an ID qualifies as a numeric string, and for anything else we throw.

Reply:
updated to a more specific type

Reply:
Re: retrieval, all the ID columns are primary key (which implies `NOT NULL`) and of integer type, so I expect that the ID should never be missing, so `from: (value: number): string => value.toString()` should work. But this breaks a ton of tests.

Re: storage, before this change, the typing ensured we were taking in a number, but with this change to strings we are now open to coercion issues so I'd expect we'd take in the broadest possible type and marshal it to a number, plus accept `undefined` for when we're autogenerating the ID. But this also breaks a ton of tests.

```ts
to: (value: unknown): number | undefined => {
  if (value === undefined) return value;

  if (typeof value !== 'string') throw new Error('Not a string');

  const asNumber = Number(value);

  if (isNaN(asNumber)) throw new Error('Not a numeric string');

  return asNumber;
},
```

Rather confusing. Maybe best to dig into this in later.